wormhole-connect/src/views/v2/TxHistory/Widget/Item.tsx

```diff
@@ -0,0 +1,281 @@
+import Box from '@mui/material/Box';
+import Card from '@mui/material/Card';
+import CardActionArea from '@mui/material/CardActionArea';
+import CardContent from '@mui/material/CardContent';
+import CircularProgress from '@mui/material/CircularProgress';
+import LinearProgress from '@mui/material/LinearProgress';
+import Stack from '@mui/material/Stack';
+import Typography from '@mui/material/Typography';
+import { makeStyles } from 'tss-react/mui';
+import { poll } from 'poll';
+
+import config from 'config';
+import { LOCAL_STORAGE_TX } from 'config/constants';
+import ArrowRight from 'icons/ArrowRight';
+import TokenIcon from 'icons/TokenIcons';
+import TxCompleteIcon from 'icons/TxComplete';
+import { minutesAndSecondsWithPadding } from 'utils/transferValidation';
+import { toFixedDecimals } from 'utils/balance';
+
+import type { TransactionLocal } from 'config/types';
+
+const useStyles = makeStyles()((theme: any) => ({
+  arrowIcon: {
+    fontSize: '16px',
+    margin: '0 4px',
+  },
+  card: {
+    width: '100%',
+    boxShadow: `0px 0px 3.5px 0px ${theme.palette.primary.main}`,
+  },
+  chainIcon: {
+    background: theme.palette.background.default,
+    border: `2px solid ${theme.palette.modal.background}`,
+    borderRadius: '6px',
+    padding: '2px',
+  },
+  completedIcon: { height: '24px', width: '24px' },
+  container: {
+    width: '100%',
+    maxWidth: '420px',
+  },
+  progressBar: {
+    borderRadius: '4px',
+    marginTop: '8px',
+  },
+}));
+
+type Props = {
+  data: TransactionLocal;
+};
+
+const WidgetItem = (props: Props) => {
+  const [senderTimestamp, setSenderTimestamp] = useState('');
+  const [etaExpired, setEtaExpired] = useState(false);
+  const [inProgress, setInProgress] = useState(true);
+  const [isFetching, setIsFetching] = useState(false);
+
+  const { classes } = useStyles();
+
+  const { data: transaction } = props;
+  const {
+    txHash,
+    eta,
+    explorerInfo,
+    amount,
+    sourceChain,
+    destChain,
+    tokenKey,
+  } = transaction;
+
+  // Initialize the countdown
+  const { seconds, minutes, totalSeconds, isRunning, restart } = useTimer({
+    expiryTimestamp: new Date(),
+    autoStart: false,
+    onExpire: () => setEtaExpired(true),
+  });
+
+  // Check whether the current transaction is from Mayan API
+  // We need this to parse the API response accordingly
+  const isMayanTx = useMemo(
+    () => explorerInfo.apiUrl.startsWith(config.mayanApi),
+    [explorerInfo.apiUrl],
+  );
+
+  // Side-effect to poll the transaction info from the respective API
+  useEffect(() => {
+    if (!transaction) {
+      return;
+    }
+
+    let cancelled = false;
+
+    const fetchTransaction = async () => {
+      setIsFetching(true);
+
+      try {
+        const res = await fetch(explorerInfo.apiUrl);
+        const resPayload = await res.json();
+
+        if (!cancelled) {
+          let txInProgress = true;
+          let txSenderTimestamp = '';
+          if (isMayanTx) {
```

Comment:
can we normalize these into a common data structure? you could just store a boolean called `inProgress`

I really hate adding code referencing mayan like this. I will keep nagging about this: Connect is supposed to be route-agnostic. there's no reason this widget should have to know what Mayan is.

Reply:
I think it needs to know about Mayan at some level, if we want to parse the completed status from its API response.